### PR TITLE
Hide log out button when auth is not enabled

### PR DESCRIPTION
**Pull request by emlun**
### Purpose

This was an oversight in #8757: the new "Log out" button is always shown in the "Actions" menu, even when authentication is not enabled.

### Testing

These manual tests have been done:

1. With authentication disabled (set username to empty): click the "Actions" menu. Verify that the "Log out" button is not shown.
2. Set a username and password and log back in. Click the "Actions" menu. Verify that the "Log out" button is shown.
3. Unset the username and save the config. Click the "Actions" menu. Verify that the "Log out" button is again not shown (do not refresh the page after saving the config).

### Screenshots

Before (3d0da5ac60ab02cf47ca5cd4fada48cc70360ec9):

Screenshot before: Log out button shown with auth not configured

After (f032f079db283967423784ec3e3aadeb6ee2f755):

Screenshot after: Log out button not shown when auth not configured

### Documentation

No changes needed.

### CODE REVIEW

File: gui/default/index.html [hunk 1]
```diff
@@ -126,7 +126,7 @@ <h3 class="panel-title">
               <li ng-if="authenticated" class="divider" aria-hidden="true"></li>
               <li ng-if="authenticated"><a href="" ng-click="advanced()"><span class="fa fa-fw fa-cogs"></span>&nbsp;<span translate>Advanced</span></a></li>
               <li ng-if="authenticated"><a href="" ng-click="logging.show()"><span class="fa fa-fw fa-wrench"></span>&nbsp;<span translate>Logs</span></a></li>
-              <li ng-if="authenticated"><a href="" ng-click="logout()"><span class="far fa-fw fa-ban"></span>&nbsp;<span translate>Log Out</span></a></li>
+              <li ng-if="authenticated && isAuthEnabled(config)"><a href="" ng-click="logout()"><span class="far fa-fw fa-ban"></span>&nbsp;<span translate>Log Out</span></a></li>
```
Comment (calmh) [inline gui/default/index.html:129]: The config we're passing is `$scope.config`, right, so we might as well not pass it as a parameter and have isAuthEnabled do the needful directly?

Comment (emlun) [reply gui/default/index.html:129]: Yeah, fair enough. The function is also called in `refreshNoAuthWarning()`, but that also just uses `$scope.config` too.

**Comment by tomasz1986** [comment]
While touching this code, can you add a divider before the "Log Out" button? It's not really related in any way to "Advanced" and "Logs".

**Comment by emlun** [comment]
> While touching this code, can you add a divider before the "Log Out" button? It's not really related in any way to "Advanced" and "Logs".

Done.

Or should we perhaps move it to the "Shutdown" and "Restart" group? Like this:

"Log out" grouped with "Shutdown" and "Restart"

One drawback of this is it can take a second before the config loads, in which case "Log out" suddenly appears and pushes the rest down if one already has the menu open. Which could make you click the wrong button if it suddenly changes just as you're about to click.

**Comment by bt90** [comment]
Could we move that group to the bottom?

**Comment by tomasz1986** [comment]
> Could we move that group to the bottom?

Maybe something like this?

```
Settings
Show ID
---divider---
Advanced
Logs
Support Bundle
---divider---
Shutdown
Restart
Log Out
```

?

**Comment by Stolka** [comment]
I think the Ban icon (`fa-ban`) should be replaced with Sign Out icon (`fa-sign-out`).

And perhaps add a divider before the Log Out button? The thing is there is no confirmation dialog when you click Restart or Shutdown, and this change to the menu order would mess with some people's muscle memory, adding a divider before Log Out button should minimize misclicking.

**Comment by emlun** [comment]
> While touching this code, can you add a divider before the "Log Out" button? It's not really related in any way to "Advanced" and "Logs".

Done.

> I think the Ban icon (`fa-ban`) should be replaced with Sign Out icon (`fa-sign-out`).

Done.

> Or should we perhaps move it to the "Shutdown" and "Restart" group? Like this:

> Could we move that group to the bottom?

> And perhaps add a divider before the Log Out button? The thing is there is no confirmation dialog when you click Restart or Shutdown, and this change to the menu order would mess with some people's muscle memory, adding a divider before Log Out button should minimize misclicking.

This seems to balloon the scope of the PR a bit too much, and I'd prefer to not hold up the original fix with an extended discussion of if and how to move things around in the menu, so I'd prefer to punt that to a separate PR/issue. Whatever change the discussion settles on, its implementation in code will take orders of magnitude less time than the discussion itself (no offense to anyone involved, that's just the reality of this kind of thing where there's no single clearly correct solution).

So, this is what the menu looks like as of commit 558ad4f84b91ad5f2c28042c7971a88f40064a6b:

actions

Thoughts on leaving this PR at that, and punting the discussion of moving things around to a separate PR/issue?

**Comment by tomasz1986** [comment]
> Thoughts on leaving this PR at that, and punting the discussion of moving things around to a separate PR/issue?

It's not a very huge deal but I would suggest moving the button (with its divider) after the Support Bundle then. Otherwise, there will still be jumping when one has debugging enabled and the button pops up later than the rest of the menu entries.

**Comment by emlun** [comment]
> I would suggest moving the button (with its divider) after the Support Bundle then. Otherwise, there will still be jumping when one has debugging enabled

Agreed, done.

actions

**Review by acolomb — APPROVED** [review]

**Comment by acolomb** [comment]
This should probably go into the next .rc as well, since it fixes an issue with new code in the upcoming release.

**Pull request merged**